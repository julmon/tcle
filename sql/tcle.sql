CREATE EXTENSION tcle;
SELECT tcle_set_passphrase('my secret passphrase');
-- Test CREATE TABLE statement and implicit casts
CREATE TABLE t (p INTEGER, n ENCRYPT_NUMERIC, l ENCRYPT_TEXT, d ENCRYPT_TIMESTAMPTZ) USING tcleam;
INSERT INTO t SELECT i, i * 2, 'Text input number '||i, '2020-06-23 22:56:50'::TIMESTAMPTZ + make_interval(days => i) FROM generate_series(1, 10) i;
SELECT COUNT(*) FROM t;
SELECT p, n, l, d FROM t WHERE n = 6;
-- Test UPDATE statement
UPDATE t SET l = 'Updated text' WHERE n = 6;
SELECT COUNT(*) FROM t WHERE l ILIKE '%updated%';
-- Test VACUM FULL
VACUUM FULL t;
SELECT * FROM t WHERE l ILIKE '%updated%';
-- Test CREATE TABLE AS statement
CREATE TABLE t2 (n, l, d) USING tcleam AS SELECT n, l, d FROM t WHERE n > 6;
SELECT * FROM t2;
-- After t2 been dropped, attached record from KMS table should have been
-- deleted
DROP TABLE t2;
SELECT COUNT(*) FROM tcle_table_keys WHERE relname = 't2';
-- Test row access by index scan
CREATE TABLE t_i (i INTEGER PRIMARY KEY, l ENCRYPT_TEXT) USING tcleam;
INSERT INTO t_i SELECT n, 'Message nÂ°'||n FROM generate_series(1, 10) n;
SET enable_seqscan TO off;
SELECT * FROM t_i WHERE i = 5;
-- Setting a new passphrase when a master key is already in use should raise
-- an error
SELECT tcle_set_passphrase('wrong secret passphrase');
