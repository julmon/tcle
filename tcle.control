# TCLE extension
comment = 'Transparent Cell-Level Encryption'
default_version = '1.0'
module_pathname = '$libdir/tcle'
relocatable = true
