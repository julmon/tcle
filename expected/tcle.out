CREATE EXTENSION tcle;
SELECT tcle_set_passphrase('my secret passphrase');
 tcle_set_passphrase 
---------------------
 t
(1 row)

-- Test CREATE TABLE statement and implicit casts
CREATE TABLE t (p INTEGER, n ENCRYPT_NUMERIC, l ENCRYPT_TEXT, d ENCRYPT_TIMESTAMPTZ) USING tcleam;
INSERT INTO t SELECT i, i * 2, 'Text input number '||i, '2020-06-23 22:56:50'::TIMESTAMPTZ + make_interval(days => i) FROM generate_series(1, 100) i;
SELECT COUNT(*) FROM t;
 count 
-------
   100
(1 row)

SELECT p, n, l, d FROM t WHERE n = 6;
 p | n |          l          |              d               
---+---+---------------------+------------------------------
 3 | 6 | Text input number 3 | Fri Jun 26 22:56:50 2020 PDT
(1 row)

-- Test UPDATE statement
UPDATE t SET l = 'Updated text' WHERE n = 6;
SELECT COUNT(*) FROM t WHERE l ILIKE '%updated%';
 count 
-------
     1
(1 row)

-- Test VACUM FULL
VACUUM FULL t;
SELECT * FROM t WHERE l ILIKE '%updated%';
 p | n |      l       |              d               
---+---+--------------+------------------------------
 3 | 6 | Updated text | Fri Jun 26 22:56:50 2020 PDT
(1 row)

-- Test CREATE TABLE AS statement
CREATE TABLE t2 (n, l, d) USING tcleam AS SELECT n, l, d FROM t WHERE n > 6;
SELECT * FROM t2;
  n  |           l           |              d               
-----+-----------------------+------------------------------
 8   | Text input number 4   | Sat Jun 27 22:56:50 2020 PDT
 10  | Text input number 5   | Sun Jun 28 22:56:50 2020 PDT
 12  | Text input number 6   | Mon Jun 29 22:56:50 2020 PDT
 14  | Text input number 7   | Tue Jun 30 22:56:50 2020 PDT
 16  | Text input number 8   | Wed Jul 01 22:56:50 2020 PDT
 18  | Text input number 9   | Thu Jul 02 22:56:50 2020 PDT
 20  | Text input number 10  | Fri Jul 03 22:56:50 2020 PDT
 22  | Text input number 11  | Sat Jul 04 22:56:50 2020 PDT
 24  | Text input number 12  | Sun Jul 05 22:56:50 2020 PDT
 26  | Text input number 13  | Mon Jul 06 22:56:50 2020 PDT
 28  | Text input number 14  | Tue Jul 07 22:56:50 2020 PDT
 30  | Text input number 15  | Wed Jul 08 22:56:50 2020 PDT
 32  | Text input number 16  | Thu Jul 09 22:56:50 2020 PDT
 34  | Text input number 17  | Fri Jul 10 22:56:50 2020 PDT
 36  | Text input number 18  | Sat Jul 11 22:56:50 2020 PDT
 38  | Text input number 19  | Sun Jul 12 22:56:50 2020 PDT
 40  | Text input number 20  | Mon Jul 13 22:56:50 2020 PDT
 42  | Text input number 21  | Tue Jul 14 22:56:50 2020 PDT
 44  | Text input number 22  | Wed Jul 15 22:56:50 2020 PDT
 46  | Text input number 23  | Thu Jul 16 22:56:50 2020 PDT
 48  | Text input number 24  | Fri Jul 17 22:56:50 2020 PDT
 50  | Text input number 25  | Sat Jul 18 22:56:50 2020 PDT
 52  | Text input number 26  | Sun Jul 19 22:56:50 2020 PDT
 54  | Text input number 27  | Mon Jul 20 22:56:50 2020 PDT
 56  | Text input number 28  | Tue Jul 21 22:56:50 2020 PDT
 58  | Text input number 29  | Wed Jul 22 22:56:50 2020 PDT
 60  | Text input number 30  | Thu Jul 23 22:56:50 2020 PDT
 62  | Text input number 31  | Fri Jul 24 22:56:50 2020 PDT
 64  | Text input number 32  | Sat Jul 25 22:56:50 2020 PDT
 66  | Text input number 33  | Sun Jul 26 22:56:50 2020 PDT
 68  | Text input number 34  | Mon Jul 27 22:56:50 2020 PDT
 70  | Text input number 35  | Tue Jul 28 22:56:50 2020 PDT
 72  | Text input number 36  | Wed Jul 29 22:56:50 2020 PDT
 74  | Text input number 37  | Thu Jul 30 22:56:50 2020 PDT
 76  | Text input number 38  | Fri Jul 31 22:56:50 2020 PDT
 78  | Text input number 39  | Sat Aug 01 22:56:50 2020 PDT
 80  | Text input number 40  | Sun Aug 02 22:56:50 2020 PDT
 82  | Text input number 41  | Mon Aug 03 22:56:50 2020 PDT
 84  | Text input number 42  | Tue Aug 04 22:56:50 2020 PDT
 86  | Text input number 43  | Wed Aug 05 22:56:50 2020 PDT
 88  | Text input number 44  | Thu Aug 06 22:56:50 2020 PDT
 90  | Text input number 45  | Fri Aug 07 22:56:50 2020 PDT
 92  | Text input number 46  | Sat Aug 08 22:56:50 2020 PDT
 94  | Text input number 47  | Sun Aug 09 22:56:50 2020 PDT
 96  | Text input number 48  | Mon Aug 10 22:56:50 2020 PDT
 98  | Text input number 49  | Tue Aug 11 22:56:50 2020 PDT
 100 | Text input number 50  | Wed Aug 12 22:56:50 2020 PDT
 102 | Text input number 51  | Thu Aug 13 22:56:50 2020 PDT
 104 | Text input number 52  | Fri Aug 14 22:56:50 2020 PDT
 106 | Text input number 53  | Sat Aug 15 22:56:50 2020 PDT
 108 | Text input number 54  | Sun Aug 16 22:56:50 2020 PDT
 110 | Text input number 55  | Mon Aug 17 22:56:50 2020 PDT
 112 | Text input number 56  | Tue Aug 18 22:56:50 2020 PDT
 114 | Text input number 57  | Wed Aug 19 22:56:50 2020 PDT
 116 | Text input number 58  | Thu Aug 20 22:56:50 2020 PDT
 118 | Text input number 59  | Fri Aug 21 22:56:50 2020 PDT
 120 | Text input number 60  | Sat Aug 22 22:56:50 2020 PDT
 122 | Text input number 61  | Sun Aug 23 22:56:50 2020 PDT
 124 | Text input number 62  | Mon Aug 24 22:56:50 2020 PDT
 126 | Text input number 63  | Tue Aug 25 22:56:50 2020 PDT
 128 | Text input number 64  | Wed Aug 26 22:56:50 2020 PDT
 130 | Text input number 65  | Thu Aug 27 22:56:50 2020 PDT
 132 | Text input number 66  | Fri Aug 28 22:56:50 2020 PDT
 134 | Text input number 67  | Sat Aug 29 22:56:50 2020 PDT
 136 | Text input number 68  | Sun Aug 30 22:56:50 2020 PDT
 138 | Text input number 69  | Mon Aug 31 22:56:50 2020 PDT
 140 | Text input number 70  | Tue Sep 01 22:56:50 2020 PDT
 142 | Text input number 71  | Wed Sep 02 22:56:50 2020 PDT
 144 | Text input number 72  | Thu Sep 03 22:56:50 2020 PDT
 146 | Text input number 73  | Fri Sep 04 22:56:50 2020 PDT
 148 | Text input number 74  | Sat Sep 05 22:56:50 2020 PDT
 150 | Text input number 75  | Sun Sep 06 22:56:50 2020 PDT
 152 | Text input number 76  | Mon Sep 07 22:56:50 2020 PDT
 154 | Text input number 77  | Tue Sep 08 22:56:50 2020 PDT
 156 | Text input number 78  | Wed Sep 09 22:56:50 2020 PDT
 158 | Text input number 79  | Thu Sep 10 22:56:50 2020 PDT
 160 | Text input number 80  | Fri Sep 11 22:56:50 2020 PDT
 162 | Text input number 81  | Sat Sep 12 22:56:50 2020 PDT
 164 | Text input number 82  | Sun Sep 13 22:56:50 2020 PDT
 166 | Text input number 83  | Mon Sep 14 22:56:50 2020 PDT
 168 | Text input number 84  | Tue Sep 15 22:56:50 2020 PDT
 170 | Text input number 85  | Wed Sep 16 22:56:50 2020 PDT
 172 | Text input number 86  | Thu Sep 17 22:56:50 2020 PDT
 174 | Text input number 87  | Fri Sep 18 22:56:50 2020 PDT
 176 | Text input number 88  | Sat Sep 19 22:56:50 2020 PDT
 178 | Text input number 89  | Sun Sep 20 22:56:50 2020 PDT
 180 | Text input number 90  | Mon Sep 21 22:56:50 2020 PDT
 182 | Text input number 91  | Tue Sep 22 22:56:50 2020 PDT
 184 | Text input number 92  | Wed Sep 23 22:56:50 2020 PDT
 186 | Text input number 93  | Thu Sep 24 22:56:50 2020 PDT
 188 | Text input number 94  | Fri Sep 25 22:56:50 2020 PDT
 190 | Text input number 95  | Sat Sep 26 22:56:50 2020 PDT
 192 | Text input number 96  | Sun Sep 27 22:56:50 2020 PDT
 194 | Text input number 97  | Mon Sep 28 22:56:50 2020 PDT
 196 | Text input number 98  | Tue Sep 29 22:56:50 2020 PDT
 198 | Text input number 99  | Wed Sep 30 22:56:50 2020 PDT
 200 | Text input number 100 | Thu Oct 01 22:56:50 2020 PDT
(97 rows)

-- After t2 been dropped, attached record from KMS table should have been
-- deleted
DROP TABLE t2;
SELECT COUNT(*) FROM tcle_table_keys WHERE relname = 't2';
 count 
-------
     0
(1 row)

-- Test row access by index scan
CREATE TABLE t_i (i INTEGER PRIMARY KEY, l ENCRYPT_TEXT) USING tcleam;
INSERT INTO t_i SELECT n, 'Message n°'||n FROM generate_series(1, 10) n;
SET enable_seqscan TO off;
SELECT * FROM t_i WHERE i = 5;
 i |      l      
---+-------------
 5 | Message n°5
(1 row)

-- Test opertor classes
SELECT * FROM t WHERE n >= 50 ORDER BY n DESC;
  p  |  n  |           l           |              d               
-----+-----+-----------------------+------------------------------
 100 | 200 | Text input number 100 | Thu Oct 01 22:56:50 2020 PDT
  99 | 198 | Text input number 99  | Wed Sep 30 22:56:50 2020 PDT
  98 | 196 | Text input number 98  | Tue Sep 29 22:56:50 2020 PDT
  97 | 194 | Text input number 97  | Mon Sep 28 22:56:50 2020 PDT
  96 | 192 | Text input number 96  | Sun Sep 27 22:56:50 2020 PDT
  95 | 190 | Text input number 95  | Sat Sep 26 22:56:50 2020 PDT
  94 | 188 | Text input number 94  | Fri Sep 25 22:56:50 2020 PDT
  93 | 186 | Text input number 93  | Thu Sep 24 22:56:50 2020 PDT
  92 | 184 | Text input number 92  | Wed Sep 23 22:56:50 2020 PDT
  91 | 182 | Text input number 91  | Tue Sep 22 22:56:50 2020 PDT
  90 | 180 | Text input number 90  | Mon Sep 21 22:56:50 2020 PDT
  89 | 178 | Text input number 89  | Sun Sep 20 22:56:50 2020 PDT
  88 | 176 | Text input number 88  | Sat Sep 19 22:56:50 2020 PDT
  87 | 174 | Text input number 87  | Fri Sep 18 22:56:50 2020 PDT
  86 | 172 | Text input number 86  | Thu Sep 17 22:56:50 2020 PDT
  85 | 170 | Text input number 85  | Wed Sep 16 22:56:50 2020 PDT
  84 | 168 | Text input number 84  | Tue Sep 15 22:56:50 2020 PDT
  83 | 166 | Text input number 83  | Mon Sep 14 22:56:50 2020 PDT
  82 | 164 | Text input number 82  | Sun Sep 13 22:56:50 2020 PDT
  81 | 162 | Text input number 81  | Sat Sep 12 22:56:50 2020 PDT
  80 | 160 | Text input number 80  | Fri Sep 11 22:56:50 2020 PDT
  79 | 158 | Text input number 79  | Thu Sep 10 22:56:50 2020 PDT
  78 | 156 | Text input number 78  | Wed Sep 09 22:56:50 2020 PDT
  77 | 154 | Text input number 77  | Tue Sep 08 22:56:50 2020 PDT
  76 | 152 | Text input number 76  | Mon Sep 07 22:56:50 2020 PDT
  75 | 150 | Text input number 75  | Sun Sep 06 22:56:50 2020 PDT
  74 | 148 | Text input number 74  | Sat Sep 05 22:56:50 2020 PDT
  73 | 146 | Text input number 73  | Fri Sep 04 22:56:50 2020 PDT
  72 | 144 | Text input number 72  | Thu Sep 03 22:56:50 2020 PDT
  71 | 142 | Text input number 71  | Wed Sep 02 22:56:50 2020 PDT
  70 | 140 | Text input number 70  | Tue Sep 01 22:56:50 2020 PDT
  69 | 138 | Text input number 69  | Mon Aug 31 22:56:50 2020 PDT
  68 | 136 | Text input number 68  | Sun Aug 30 22:56:50 2020 PDT
  67 | 134 | Text input number 67  | Sat Aug 29 22:56:50 2020 PDT
  66 | 132 | Text input number 66  | Fri Aug 28 22:56:50 2020 PDT
  65 | 130 | Text input number 65  | Thu Aug 27 22:56:50 2020 PDT
  64 | 128 | Text input number 64  | Wed Aug 26 22:56:50 2020 PDT
  63 | 126 | Text input number 63  | Tue Aug 25 22:56:50 2020 PDT
  62 | 124 | Text input number 62  | Mon Aug 24 22:56:50 2020 PDT
  61 | 122 | Text input number 61  | Sun Aug 23 22:56:50 2020 PDT
  60 | 120 | Text input number 60  | Sat Aug 22 22:56:50 2020 PDT
  59 | 118 | Text input number 59  | Fri Aug 21 22:56:50 2020 PDT
  58 | 116 | Text input number 58  | Thu Aug 20 22:56:50 2020 PDT
  57 | 114 | Text input number 57  | Wed Aug 19 22:56:50 2020 PDT
  56 | 112 | Text input number 56  | Tue Aug 18 22:56:50 2020 PDT
  55 | 110 | Text input number 55  | Mon Aug 17 22:56:50 2020 PDT
  54 | 108 | Text input number 54  | Sun Aug 16 22:56:50 2020 PDT
  53 | 106 | Text input number 53  | Sat Aug 15 22:56:50 2020 PDT
  52 | 104 | Text input number 52  | Fri Aug 14 22:56:50 2020 PDT
  51 | 102 | Text input number 51  | Thu Aug 13 22:56:50 2020 PDT
  50 | 100 | Text input number 50  | Wed Aug 12 22:56:50 2020 PDT
  49 | 98  | Text input number 49  | Tue Aug 11 22:56:50 2020 PDT
  48 | 96  | Text input number 48  | Mon Aug 10 22:56:50 2020 PDT
  47 | 94  | Text input number 47  | Sun Aug 09 22:56:50 2020 PDT
  46 | 92  | Text input number 46  | Sat Aug 08 22:56:50 2020 PDT
  45 | 90  | Text input number 45  | Fri Aug 07 22:56:50 2020 PDT
  44 | 88  | Text input number 44  | Thu Aug 06 22:56:50 2020 PDT
  43 | 86  | Text input number 43  | Wed Aug 05 22:56:50 2020 PDT
  42 | 84  | Text input number 42  | Tue Aug 04 22:56:50 2020 PDT
  41 | 82  | Text input number 41  | Mon Aug 03 22:56:50 2020 PDT
  40 | 80  | Text input number 40  | Sun Aug 02 22:56:50 2020 PDT
  39 | 78  | Text input number 39  | Sat Aug 01 22:56:50 2020 PDT
  38 | 76  | Text input number 38  | Fri Jul 31 22:56:50 2020 PDT
  37 | 74  | Text input number 37  | Thu Jul 30 22:56:50 2020 PDT
  36 | 72  | Text input number 36  | Wed Jul 29 22:56:50 2020 PDT
  35 | 70  | Text input number 35  | Tue Jul 28 22:56:50 2020 PDT
  34 | 68  | Text input number 34  | Mon Jul 27 22:56:50 2020 PDT
  33 | 66  | Text input number 33  | Sun Jul 26 22:56:50 2020 PDT
  32 | 64  | Text input number 32  | Sat Jul 25 22:56:50 2020 PDT
  31 | 62  | Text input number 31  | Fri Jul 24 22:56:50 2020 PDT
  30 | 60  | Text input number 30  | Thu Jul 23 22:56:50 2020 PDT
  29 | 58  | Text input number 29  | Wed Jul 22 22:56:50 2020 PDT
  28 | 56  | Text input number 28  | Tue Jul 21 22:56:50 2020 PDT
  27 | 54  | Text input number 27  | Mon Jul 20 22:56:50 2020 PDT
  26 | 52  | Text input number 26  | Sun Jul 19 22:56:50 2020 PDT
  25 | 50  | Text input number 25  | Sat Jul 18 22:56:50 2020 PDT
(76 rows)

SELECT * FROM t WHERE l < 'Text input number 3' ORDER BY l DESC;
  p  |  n  |           l           |              d               
-----+-----+-----------------------+------------------------------
  29 | 58  | Text input number 29  | Wed Jul 22 22:56:50 2020 PDT
  28 | 56  | Text input number 28  | Tue Jul 21 22:56:50 2020 PDT
  27 | 54  | Text input number 27  | Mon Jul 20 22:56:50 2020 PDT
  26 | 52  | Text input number 26  | Sun Jul 19 22:56:50 2020 PDT
  25 | 50  | Text input number 25  | Sat Jul 18 22:56:50 2020 PDT
  24 | 48  | Text input number 24  | Fri Jul 17 22:56:50 2020 PDT
  23 | 46  | Text input number 23  | Thu Jul 16 22:56:50 2020 PDT
  22 | 44  | Text input number 22  | Wed Jul 15 22:56:50 2020 PDT
  21 | 42  | Text input number 21  | Tue Jul 14 22:56:50 2020 PDT
  20 | 40  | Text input number 20  | Mon Jul 13 22:56:50 2020 PDT
   2 | 4   | Text input number 2   | Thu Jun 25 22:56:50 2020 PDT
  19 | 38  | Text input number 19  | Sun Jul 12 22:56:50 2020 PDT
  18 | 36  | Text input number 18  | Sat Jul 11 22:56:50 2020 PDT
  17 | 34  | Text input number 17  | Fri Jul 10 22:56:50 2020 PDT
  16 | 32  | Text input number 16  | Thu Jul 09 22:56:50 2020 PDT
  15 | 30  | Text input number 15  | Wed Jul 08 22:56:50 2020 PDT
  14 | 28  | Text input number 14  | Tue Jul 07 22:56:50 2020 PDT
  13 | 26  | Text input number 13  | Mon Jul 06 22:56:50 2020 PDT
  12 | 24  | Text input number 12  | Sun Jul 05 22:56:50 2020 PDT
  11 | 22  | Text input number 11  | Sat Jul 04 22:56:50 2020 PDT
 100 | 200 | Text input number 100 | Thu Oct 01 22:56:50 2020 PDT
  10 | 20  | Text input number 10  | Fri Jul 03 22:56:50 2020 PDT
   1 | 2   | Text input number 1   | Wed Jun 24 22:56:50 2020 PDT
(23 rows)

SELECT * FROM t WHERE d > '2020-08-01 00:00:00' ORDER BY d DESC;
  p  |  n  |           l           |              d               
-----+-----+-----------------------+------------------------------
 100 | 200 | Text input number 100 | Thu Oct 01 22:56:50 2020 PDT
  99 | 198 | Text input number 99  | Wed Sep 30 22:56:50 2020 PDT
  98 | 196 | Text input number 98  | Tue Sep 29 22:56:50 2020 PDT
  97 | 194 | Text input number 97  | Mon Sep 28 22:56:50 2020 PDT
  96 | 192 | Text input number 96  | Sun Sep 27 22:56:50 2020 PDT
  95 | 190 | Text input number 95  | Sat Sep 26 22:56:50 2020 PDT
  94 | 188 | Text input number 94  | Fri Sep 25 22:56:50 2020 PDT
  93 | 186 | Text input number 93  | Thu Sep 24 22:56:50 2020 PDT
  92 | 184 | Text input number 92  | Wed Sep 23 22:56:50 2020 PDT
  91 | 182 | Text input number 91  | Tue Sep 22 22:56:50 2020 PDT
  90 | 180 | Text input number 90  | Mon Sep 21 22:56:50 2020 PDT
  89 | 178 | Text input number 89  | Sun Sep 20 22:56:50 2020 PDT
  88 | 176 | Text input number 88  | Sat Sep 19 22:56:50 2020 PDT
  87 | 174 | Text input number 87  | Fri Sep 18 22:56:50 2020 PDT
  86 | 172 | Text input number 86  | Thu Sep 17 22:56:50 2020 PDT
  85 | 170 | Text input number 85  | Wed Sep 16 22:56:50 2020 PDT
  84 | 168 | Text input number 84  | Tue Sep 15 22:56:50 2020 PDT
  83 | 166 | Text input number 83  | Mon Sep 14 22:56:50 2020 PDT
  82 | 164 | Text input number 82  | Sun Sep 13 22:56:50 2020 PDT
  81 | 162 | Text input number 81  | Sat Sep 12 22:56:50 2020 PDT
  80 | 160 | Text input number 80  | Fri Sep 11 22:56:50 2020 PDT
  79 | 158 | Text input number 79  | Thu Sep 10 22:56:50 2020 PDT
  78 | 156 | Text input number 78  | Wed Sep 09 22:56:50 2020 PDT
  77 | 154 | Text input number 77  | Tue Sep 08 22:56:50 2020 PDT
  76 | 152 | Text input number 76  | Mon Sep 07 22:56:50 2020 PDT
  75 | 150 | Text input number 75  | Sun Sep 06 22:56:50 2020 PDT
  74 | 148 | Text input number 74  | Sat Sep 05 22:56:50 2020 PDT
  73 | 146 | Text input number 73  | Fri Sep 04 22:56:50 2020 PDT
  72 | 144 | Text input number 72  | Thu Sep 03 22:56:50 2020 PDT
  71 | 142 | Text input number 71  | Wed Sep 02 22:56:50 2020 PDT
  70 | 140 | Text input number 70  | Tue Sep 01 22:56:50 2020 PDT
  69 | 138 | Text input number 69  | Mon Aug 31 22:56:50 2020 PDT
  68 | 136 | Text input number 68  | Sun Aug 30 22:56:50 2020 PDT
  67 | 134 | Text input number 67  | Sat Aug 29 22:56:50 2020 PDT
  66 | 132 | Text input number 66  | Fri Aug 28 22:56:50 2020 PDT
  65 | 130 | Text input number 65  | Thu Aug 27 22:56:50 2020 PDT
  64 | 128 | Text input number 64  | Wed Aug 26 22:56:50 2020 PDT
  63 | 126 | Text input number 63  | Tue Aug 25 22:56:50 2020 PDT
  62 | 124 | Text input number 62  | Mon Aug 24 22:56:50 2020 PDT
  61 | 122 | Text input number 61  | Sun Aug 23 22:56:50 2020 PDT
  60 | 120 | Text input number 60  | Sat Aug 22 22:56:50 2020 PDT
  59 | 118 | Text input number 59  | Fri Aug 21 22:56:50 2020 PDT
  58 | 116 | Text input number 58  | Thu Aug 20 22:56:50 2020 PDT
  57 | 114 | Text input number 57  | Wed Aug 19 22:56:50 2020 PDT
  56 | 112 | Text input number 56  | Tue Aug 18 22:56:50 2020 PDT
  55 | 110 | Text input number 55  | Mon Aug 17 22:56:50 2020 PDT
  54 | 108 | Text input number 54  | Sun Aug 16 22:56:50 2020 PDT
  53 | 106 | Text input number 53  | Sat Aug 15 22:56:50 2020 PDT
  52 | 104 | Text input number 52  | Fri Aug 14 22:56:50 2020 PDT
  51 | 102 | Text input number 51  | Thu Aug 13 22:56:50 2020 PDT
  50 | 100 | Text input number 50  | Wed Aug 12 22:56:50 2020 PDT
  49 | 98  | Text input number 49  | Tue Aug 11 22:56:50 2020 PDT
  48 | 96  | Text input number 48  | Mon Aug 10 22:56:50 2020 PDT
  47 | 94  | Text input number 47  | Sun Aug 09 22:56:50 2020 PDT
  46 | 92  | Text input number 46  | Sat Aug 08 22:56:50 2020 PDT
  45 | 90  | Text input number 45  | Fri Aug 07 22:56:50 2020 PDT
  44 | 88  | Text input number 44  | Thu Aug 06 22:56:50 2020 PDT
  43 | 86  | Text input number 43  | Wed Aug 05 22:56:50 2020 PDT
  42 | 84  | Text input number 42  | Tue Aug 04 22:56:50 2020 PDT
  41 | 82  | Text input number 41  | Mon Aug 03 22:56:50 2020 PDT
  40 | 80  | Text input number 40  | Sun Aug 02 22:56:50 2020 PDT
  39 | 78  | Text input number 39  | Sat Aug 01 22:56:50 2020 PDT
(62 rows)

-- Setting a new passphrase when a master key is already in use should raise
-- an error
SELECT tcle_set_passphrase('wrong secret passphrase');
ERROR:  tcle: another database master key is in use
